### PR TITLE
Ruby on Rails capstone project - Budget app

### PR DESCRIPTION
**Pull request by chudisoft**
Project requirements

-  Set up the repository and tools.
-  Create the models according to the ERD diagram. Don't forget about validations and tests.
-  Implement the authentication and authorization.
-  Create the page to add a category.
-  Create the home page.
-  Create the page to add a transaction.
-  Create the transactions page.
-  Make sure your app is tested adequately.
-  Deploy the project, and test for final details.
   https://budget-tracker-rv3z.onrender.com/
-  Record a video for your project.
   https://www.loom.com/share/8b8f8f263fd046399adbc5df94eaec35
-  Create a good README and PR description.

ERD diagram
image

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -1 +1,225 @@
...
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+
+### Presentation
+
+This project is presented at:
+```
+https://drive.google.com/file/d/1V8ybN1ztHb2e9PIVLubO7Ry4Y_xjNL7I/view?usp=sharing
+```
```
Comment (Ibizugbe) [inline Readme.md]: Nice job so far👍🏼. As a requirement, kindly ensure you talk about a piece of code/feature you've implemented that you're proud of as this also makes your presentation more engaging👍🏼.

image

File: Readme.md [hunk 2]
```diff
@@ -1 +1,225 @@
...
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+
+### Deploy
+
+This project is deployed at:
+```
+https://budget-manager1.onrender.com/
+```
```
Comment (Ibizugbe) [inline Readme.md]: Nice job deploying your application👍🏼. For an improved user experience, kindly ensure every part of the application works fine on the deployed version. currently, the signup feature doesn't work properly. please do well to fix that👍🏼.

image

File: app/views/devise/sessions/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        <%= f.password_field :password, autocomplete: "current-password", placeholder: "Password", class: 'form-control' %>
+      </div>
+      <%= f.input :remember_me, as: :boolean if devise_mapping.rememberable? %>
+    </div>
+
+    <div class="card-footer p-2">
+      <div class="form-group">
+        <%= f.button :submit, "Log in", class: 'btn btn-primary w-100' %>
+      </div>
+    </div>
```
Comment (Ibizugbe) [inline app/views/devise/sessions/new.html.erb]: Nice job so far👍🏼. Let's ensure the login/signup buttons match the [design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=)😉.

yours 
image


design
Screenshot 2024-02-05 at 08 49 56

please take note of the space left and right of the login button😉.

File: app/views/entities/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+              <% end %>
+          <% end %>
+        </div>
+      </div>
+    <% end %>
+  </div>
+</div>
+  <div class="text-center bottom-element w-100">
+    <%= link_to "New Entity", new_group_entity_path(@group), class: "btn btn-success w-100" %>
+  </div>
```
Comment (Ibizugbe) [inline app/views/entities/index.html.erb]: kindly do well to also stick to the design here, Kindly implement your `new entity/group` buttons to match the given design😉. take not of the border lines, the padding and margin.

yours
image

design
image

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @chudisoft 👋🏼👋🏼👋🏼, 

request change


Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights✅✅✅

- [x] Linters are passing👍🏼.
- [x] Input validations are implemented👍🏼.
- [x] Nice job with the "recent" and "oldest" sorting feature👍🏼.
- [x] Tests are written and they are passing👍🏼.
- [x] Nice job with the delete and edit feature for each group and transaction👍🏼.


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- [OPTIONAL] For your PR description, I suggest you always outline the features you have implemented in the branch, instead of the requirements of the project😉.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by prolajumokeoni — APPROVED** [review]
Hi @chudisoft 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
 - Deployed link works properly ✅ 
 - Updated video presentation 👏🏿 
 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- It would be nice to reorder your transactions( entities) by the most recent,  from the requirements the list of transactions is presented ordered by the most recent, this will be  a nice to have feature as you have implemented something like this for your categories(groups) 👍🏿 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**